plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/_core_aws.py

```diff
@@ -96,39 +106,41 @@ def launch(
             task_function=launcher.task_function,
             singleton_state=Singleton.get_state(),
         )
-
-        with tempfile.NamedTemporaryFile(suffix=".yaml", delete=False) as f:
-            with open(f.name, "w") as file:
-                OmegaConf.save(
-                    config=launcher.ray_cfg.cluster, f=file.name, resolve=True
-                )
-            launcher.ray_yaml_path = f.name
-            log.info(
-                f"Saving RayClusterConf in a temp yaml file: {launcher.ray_yaml_path}."
-            )
-
-            return launch_jobs(
-                launcher, local_tmp_dir, Path(launcher.config.hydra.sweep.dir)
-            )
+        return launch_jobs(
+            launcher, local_tmp_dir, Path(launcher.config.hydra.sweep.dir)
+        )
 
 
 def launch_jobs(
     launcher: RayAWSLauncher, local_tmp_dir: str, sweep_dir: Path
 ) -> Sequence[JobReturn]:
-    ray_up(launcher.ray_yaml_path)
+    config = OmegaConf.to_container(
+        launcher.ray_cfg.cluster, resolve=True, enum_to_str=True
+    )
+    sdk.create_or_update_cluster(  # type: ignore[attr-defined]
+        config,
+        no_restart=launcher.update_cluster_no_restart,
+        restart_only=launcher.update_cluster_restart_only,
+        no_config_cache=launcher.update_cluster_no_config_cache,
```

Comment:
Is it possible to fold these 3 fields into the cluster config, just like how we are planning on grouping the logging config?